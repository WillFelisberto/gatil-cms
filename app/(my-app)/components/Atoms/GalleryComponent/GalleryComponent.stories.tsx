import type { Meta, StoryObj } from '@storybook/react';

import type { BlockProps } from '@/app/(my-app)/types/block-props';

import { GalleryComponent } from './GalleryComponent';

const meta: Meta<typeof GalleryComponent> = {
  title: 'Molecules/GalleryComponent',
  component: GalleryComponent,
  tags: ['autodocs']
};

export default meta;

type Story = StoryObj<typeof GalleryComponent>;

export const Default: Story = {
  args: {
    blockType: 'gallery',
    images: [
      {
        image: { url: '/gato.jpg', alt: 'Gato preto' },
        caption: 'Gato preto'
      },
      {
        image: { url: '/gato.jpg', alt: 'Gato branco' },
        caption: 'Gato branco'
      },
      {
        image: { url: '/gato.jpg', alt: 'Gato cinza' },
        caption: 'Gato cinza'
      }
    ]
  } as BlockProps<'gallery'>
};
