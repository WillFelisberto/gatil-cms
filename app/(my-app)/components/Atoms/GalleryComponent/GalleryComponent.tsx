'use client';

import Image from 'next/image';

import type { BlockProps } from '@/app/(my-app)/types/block-props';
import type { Media } from '@/payload-types';

export const GalleryComponent = (props: BlockProps<'gallery'>) => {
  const { images } = props;

  return (
    <section data-testid="gallery-block" className="w-full px-4 py-12">
      <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-6 max-w-7xl mx-auto">
        {images?.map(({ image, caption }, i) => {
          const img = image as Media;

          if (!img?.url) return null;

          return (
            <figure key={img.url || i} className="rounded overflow-hidden shadow-sm">
              <div className="relative w-full h-64">
                <Image
                  src={img.url}
                  alt={img.alt || caption || `Imagem ${i + 1}`}
                  fill
                  className="object-cover rounded"
                  data-testid="gallery-image"
                  sizes="(max-width: 768px) 100vw, (max-width: 1024px) 50vw, 33vw"
                  priority={i < 3} // prioridade para primeiras imagens
                />
              </div>
              {caption && (
                <figcaption
                  className="text-sm text-center text-gray-700 mt-2"
                  data-testid="gallery-caption"
                >
                  {caption}
                </figcaption>
              )}
            </figure>
          );
        })}
      </div>
    </section>
  );
};
