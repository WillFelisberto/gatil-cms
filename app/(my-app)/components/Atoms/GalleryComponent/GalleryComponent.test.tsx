import type { BlockProps } from '@/app/(my-app)/types/block-props';
import { render, screen } from '@/tests/test-utils';

import { GalleryComponent } from './GalleryComponent';

const mockGallery: BlockProps<'gallery'> = {
  blockType: 'gallery',
  images: [
    {
      image: {
        id: '1',
        url: '/gato1.jpg',
        alt: 'Gato dormindo',
        createdAt: '2023-01-01',
        updatedAt: '2023-01-02'
      },
      caption: 'Gato dormindo'
    },
    {
      image: {
        id: '2',
        url: '/gato2.jpg',
        alt: 'Gato brincando',
        createdAt: '2023-01-01',
        updatedAt: '2023-01-02'
      },
      caption: 'Gato brincando'
    }
  ]
};

describe('<GalleryComponent />', () => {
  it('Should renders all gallery images and captions', () => {
    const { container } = render(<GalleryComponent {...mockGallery} />);

    const images = screen.getAllByTestId('gallery-image');
    expect(images).toHaveLength(2);

    expect(screen.getByText('Gato dormindo')).toBeInTheDocument();
    expect(screen.getByText('Gato brincando')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('Should renders the section container', () => {
    render(<GalleryComponent {...mockGallery} />);
    expect(screen.getByTestId('gallery-block')).toBeInTheDocument();
  });
});
