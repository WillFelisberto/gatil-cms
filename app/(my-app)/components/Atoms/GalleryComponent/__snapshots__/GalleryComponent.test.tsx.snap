// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GalleryComponent /> Should renders all gallery images and captions 1`] = `
<div>
  <section
    class="w-full px-4 py-12"
    data-testid="gallery-block"
  >
    <div
      class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-6 max-w-7xl mx-auto"
    >
      <figure
        class="rounded overflow-hidden shadow-sm"
      >
        <div
          class="relative w-full h-64"
        >
          <img
            alt="Gato dormindo"
            class="object-cover rounded"
            data-nimg="fill"
            data-testid="gallery-image"
            decoding="async"
            sizes="(max-width: 768px) 100vw, (max-width: 1024px) 50vw, 33vw"
            src="/_next/image?url=%2Fgato1.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fgato1.jpg&w=256&q=75 256w, /_next/image?url=%2Fgato1.jpg&w=384&q=75 384w, /_next/image?url=%2Fgato1.jpg&w=640&q=75 640w, /_next/image?url=%2Fgato1.jpg&w=750&q=75 750w, /_next/image?url=%2Fgato1.jpg&w=828&q=75 828w, /_next/image?url=%2Fgato1.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fgato1.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fgato1.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fgato1.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fgato1.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <figcaption
          class="text-sm text-center text-gray-700 mt-2"
          data-testid="gallery-caption"
        >
          Gato dormindo
        </figcaption>
      </figure>
      <figure
        class="rounded overflow-hidden shadow-sm"
      >
        <div
          class="relative w-full h-64"
        >
          <img
            alt="Gato brincando"
            class="object-cover rounded"
            data-nimg="fill"
            data-testid="gallery-image"
            decoding="async"
            sizes="(max-width: 768px) 100vw, (max-width: 1024px) 50vw, 33vw"
            src="/_next/image?url=%2Fgato2.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fgato2.jpg&w=256&q=75 256w, /_next/image?url=%2Fgato2.jpg&w=384&q=75 384w, /_next/image?url=%2Fgato2.jpg&w=640&q=75 640w, /_next/image?url=%2Fgato2.jpg&w=750&q=75 750w, /_next/image?url=%2Fgato2.jpg&w=828&q=75 828w, /_next/image?url=%2Fgato2.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fgato2.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fgato2.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fgato2.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fgato2.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <figcaption
          class="text-sm text-center text-gray-700 mt-2"
          data-testid="gallery-caption"
        >
          Gato brincando
        </figcaption>
      </figure>
    </div>
  </section>
</div>
`;
